src/main/java/com/pinterest/secor/parser/ThriftMessageParser.java

```diff
@@ -52,10 +56,21 @@ public String getFieldName() {
         }
     }
 
-    public ThriftMessageParser(SecorConfig config) {
+    public ThriftMessageParser(SecorConfig config)
+            throws InstantiationException, IllegalAccessException,
+            ClassNotFoundException {
         super(config);
-        mDeserializer = new TDeserializer();
-        mThriftPath = new ThriftPath(mConfig.getMessageTimestampName(), (short) mConfig.getMessageTimestampId());
+        TProtocolFactory protocolFactory = null;
+        String protocolName = mConfig.getThriftProtocolClass();
+        
+        if (StringUtils.isNotEmpty(protocolName)) {
+            String factoryClassName = protocolName.concat("$Factory");
+            protocolFactory = ((Class<? extends TProtocolFactory>) Class.forName(factoryClassName)).newInstance();
+        } else
+            protocolFactory = new TBinaryProtocol.Factory();
```

Comment:
it is done under the hood:
https://github.com/apache/thrift/blob/0.9.3/lib/java/src/org/apache/thrift/TDeserializer.java

it looks good to me to be verbose in case someone ignores the default factory